sway-core/src/semantic_analysis/node_dependencies.rs

```diff
@@ -337,6 +337,7 @@ impl Dependencies {
                 })
                 .gather_from_type_parameters(type_parameters)
             }
+            Declaration::EnumVariantDeclaration(_decl) => unreachable!(),
```

Comment:
Won't we hit the `EnumVariantDeclaration` case when the `EnumDeclaration` calls `self.gater_from_iter(variants.iter(), ...)`?